### PR TITLE
VIM-4918: Default Privacy Migration

### PR DESCRIPTION
**Pull request by ghking**
#### Ticket

https://vimean.atlassian.net/browse/VIM-4918

#### Ticket Summary

Write a migration to upgrade the old version of privacy (a string) to the new version (an object).

#### Implementation Summary

Added the migration to create a new privacy object and set the string in the correct place. Jennifer already updated the model and code in another PR: https://github.vimeows.com/MobileApps/Vimeo-iOS/pull/1177

#### How to Test

- Log out
- Download the app store version
- Log in
- Upgrade
- Go to upload a video

The app should no longer crash at this point.

### CODE REVIEW

File: VimeoNetworking/Sources/Models/VIMVideoPreference.m [hunk 1]
```diff
@@ -26,15 +26,26 @@
...
-
 @implementation VIMVideoPreference
 
 #pragma mark - VIMMappable
 
+- (void)didFinishMapping
+{
+    // In API version 3.3.1, privacy changed from a string to a VIMPrivacy object.
+    // This is a migration to convert pre-3.3.1 versions to the new version. [ghking] 2/23/17
+
```
Comment (jenniferlim) [inline VimeoNetworking/Sources/Models/VIMVideoPreference.m]: nit: Do we need to precise that before 3.3.1 it was a `String`?

**Comment by jenniferlim** [comment]
@ghking  - Look great 💯 💯 💯

File: VimeoNetworking/Sources/Models/VIMVideoPreference.m [hunk 2]
```diff
@@ -26,15 +26,26 @@
...
+        VIMPrivacy *privacy = [VIMPrivacy new];
+        privacy.view = self.privacy;
+        self.privacy = privacy;
+    }
+}
+
 - (Class)getClassForObjectKey:(NSString *)key
 {
-    if ([key isEqualToString:kPrivacy])
+    if ([key isEqualToString:@"privacy"])
```
Comment (supperspidey) [inline VimeoNetworking/Sources/Models/VIMVideoPreference.m]: Can we make `@"privacy"` as a constant? I believe @jenniferlim already made the string as a constant.

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/fd9c80bcae419890c95071ec4b5f2abcd97b2263/VimeoNetworking/Sources/VimeoClient.swift#L328">VimeoNetworking/Sources/VimeoClient.swift#L328</a></strong>: cast from ‘Response&lt;VIMNullResponse&gt;’ to unrelated type ‘Response&lt;ModelType&gt;’ always fails  <br /><code>
                let response = Response(model: nullResponseObject, json: [:]) as! Response&lt;ModelType&gt;
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/fd9c80bcae419890c95071ec4b5f2abcd97b2263/VimeoNetworking/Sources/Models/VIMVideoPreference.m#L41">VimeoNetworking/Sources/Models/VIMVideoPreference.m#L41</a></strong>: incompatible pointer types assigning to ‘NSString * _Nullable’ from ‘VIMPrivacy * _Nullable’ [-Wincompatible-pointer-types]  <br /><code>
        privacy.view = self.privacy;
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 22 tests, with 0 failures (0 unexpected) in 0.292 (0.304) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/169)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Pull request merged**